lib/channel_helper.ml

```diff
@@ -112,23 +110,22 @@ let advertise_t common_options_t proxy_socket =
     Printf.fprintf stdout "%s\n%!" (Jsonrpc.to_string (Xcp_channel.rpc_of_protocols protocols));
 
   let t_ip =
-    lwt fd, peer = Lwt_unix.accept s_ip in
-    lwt () = Lwt_unix.close s_ip in
+    Lwt_unix.accept s_ip >>= fun (fd, peer) ->
+    Lwt_unix.close s_ip >>= fun () ->
     proxy fd (Lwt_unix.of_unix_file_descr proxy_socket) in
   let t_unix =
-    lwt fd, peer = Lwt_unix.accept s_unix in
+    Lwt_unix.accept s_unix >>= fun (fd, peer) ->
```

Comment:
It seems that `Lwt_unix.accept` returns a pair, so sadly we cannot change that if I'm not mistaken:
```ocaml
val accept : file_descr -> (file_descr * sockaddr) Lwt.t
```